libs/@local/hash-graph-client/python/graph_client/__init__.py

```diff
@@ -67,29 +72,56 @@ async def _send_request(
     return response_t.model_validate(json, strict=False)
 
 
+U = TypeVar("U")
+
+
+def _assert_not_none(value: U | None) -> U:
+    """Assert that the value is not None."""
+    if value is None:
+        msg = "value cannot be None"
+        raise ValueError(msg)
+
+    return value
+
+
 class GraphClient:
     """Low-level implementation of the client for the HASH API."""
 
     base: URL
+    actor: UUID | None
 
-    def __init__(self, base: URL) -> None:
+    def __init__(self, base: URL, *, actor: UUID | None = None) -> None:
```

Comment:
Just so you're aware of it: In `hash-api` we always set the `actor` on creation. I like the idea of being able to override it when calling it, I just don't know a good reason why this is useful (for `actor`).

Reply:
This is to support the two modes of operation: ad-hoc `actor` (per request) and persistent `actor`.

Both have their use cases. Let's say you have a long-running session with a single actor; in that case, you can specify the actor when creating (or setting it with the property setter). Another application might favor putting it on a per-request basis but requires the same client (the graph type protocol, for example). Both are valid use cases, so I included both in a simple API, enabling the overrides.

To bring it into some more practical terms: setting via `__init__` is something a script might do, while overrides per request basis would be done in e.g. a REST API that reuses the same client.

Reply:
Yeah, I get the two modes, but I don't see the ad-hoc mode as very useful. e.g. a REST API should better create a new client per session (= tied to a user) than having a global variable to call it every time. Passing the actor in an ad-hoc application is error-prone as it's not required and might be forgotten easily. 🤷🏻‍♂️

Reply:
To be fair, I thought you found the non-ad-hoc mode problematic :P tbf I mostly added it to accommodate the protocol used in the `hash-graph-types` which needs the ad-hoc mode. (I initially planned to have no ad-hoc mode).

Reply:
If you'd also prefer the non-ad-hoc mode let's try to remove the ad-hoc mode at some point 🙂